### ISSUE TITLE
Timelock Windows Compat

### ISSUE THREAD

**Issue by clockfort** [palantir/atlasdb#2018]
I noticed some altasdb-config tests were accidentally windows-compat broken, but as our only consistent Windows dev, I don't really care, so no big deal.

However, we do actually need our non-test code to work in Windows, and from some fairly arcane grepping for the same problem, I strongly suspect that TimeLock can't run on windows.
There are a bunch of places where there are hardcoded paths with the wrong directory separator character hardcoded into strings.

```
(not an exhaustive list)

timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/AtomixConfiguration.java:        return "var/data/atomix";
timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/PaxosConfiguration.java:        return new File("var/data/paxos");
timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimeLockConstants.java:    public static final String DEFAULT_LOG_DIRECTORY = "var/data/";
```

I think this is a low-priority blocker for {big internal project} moving onto timelock instead of their more custom lock server. They also don't run automated testing on Windows, so after they moved to using timelock proper they'd only figure out at the very end before a release that it didn't work, when they test the Windows component manually, which would really be annoying for everyone.

I can own fixing this if you guys want, considering you guys are allergic to Windows and it's somewhat low priority since {PRODUCT} hasn't moved to timelock yet.

**Comment by tpetracca** [comment]
@clockfort a bunch of the libs we depend on (like http-remoting) do this very same thing for default values of things.

**Comment by clockfort** [comment]
It's okay in HTTP URIs, but not in things that we expect to be a file path

**Comment by hsaraogi** [comment]
@clockfort This is low priority for us right now. Happy for you to fix it up!

**Comment by clockfort** [comment]
closing.
The issues were all test-only,which I have addressed in #2059
